src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -70,8 +70,16 @@ internal class SentryMiddleware
         /// <param name="context">The context.</param>
         /// <returns></returns>
         public async Task InvokeAsync(HttpContext context)
-        {
+        {           
             var hub = _hubAccessor();
+            if (_options.FlushOnCompletedRequest)
+            {
+                context.Response.OnCompleted(async () =>
+                {
+                    await _hubAccessor().FlushAsync(timeout: _options.FlushTimeout).ConfigureAwait(false);
```

Comment:
Looks like the hub was already retrieved just a few lines above
```suggestion
                    await hub.FlushAsync(timeout: _options.FlushTimeout).ConfigureAwait(false);
```